lib/taskLogger.ts

```diff
@@ -0,0 +1,92 @@
+import {EOL} from 'os';
+import * as Logger from './logger';
+
+export default class TaskLogger {
+  private task: any;
+  private pid: number;
+  private buffer: string;
+  private insertTag: boolean;
+
+  /**
+   * Log output such that metadata are appended.
+   * Calling log(data) will not flush to console until you call flush()
+   *
+   * @constructor
+   * @param {object} task Task that is being reported.
+   * @param {number} pid PID of process running the task.
+   */
+  constructor(task: any, pid: number) {
+    this.task = task;
+    this.pid = pid;
+    this.buffer = '';
```

Comment:
assign the default value at the declaration site - it is called immediately following the constructor